scalyr_agent/util.py

```diff
@@ -406,6 +406,7 @@ def remove_newlines_and_truncate(input_string, char_limit):
     @return:  The string with all newlines replaced with spaces and truncated.
     @rtype: str
     """
+    input_string = six.ensure_text(input_string)
```

Comment:
FYI:

```bash
python3 /usr/sbin/scalyr-agent-2 --no-fork start

Traceback (most recent call last):
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 464, in __receive_response
    response_as_json = scalyr_util.json_decode(response)
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 179, in json_decode
    return _json_decode(text)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/sbin/scalyr-agent-2", line 526, in __start
    self.__perform_config_checks(no_check_remote)
  File "/usr/sbin/scalyr-agent-2", line 450, in __perform_config_checks
    ping_result = client.ping()
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 237, in ping
    return self.send(self.add_events_request())[0]
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 572, in send
    "/addEvents", body_func=generate_body, block_on_response=block_on_response,
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 381, in __send_request
    return receive_response()
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 376, in receive_response
    return self.__receive_response(body_str, current_time)
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/scalyr_client.py", line 470, in __receive_response
    scalyr_util.remove_newlines_and_truncate(response, 1000),
  File "/home/vagrant/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 409, in remove_newlines_and_truncate
    return input_string.replace("\n", " ").replace("\r", " ")[0:char_limit]
TypeError: a bytes-like object is required, not 'str'
Terminating agent, please fix the error and restart the agent.
```

Reply:
Can you update the type hint for `input_str`?  I would suggest we explicitly say it can be either `six.text_type` or `six.binary_type`.

Reply:
Yeah, I will update the docstring and add type annotation.

On that note, that's also where MyPy would and will come very handy.

If we had type annotations for that function and most of the other code, MyPy would detect such issues so we could probably enforce and stick with a single strict type (bytes).

But without that, this change (aka accepting both types) is probably the most reasonable right now.

Reply:
I've updated docstring and added type annotation in e529d8b.

This may require some small adjustments once #377 is merged in and MyPy check is actually hooked up.